chat-widget/src/components/livechatwidget/common/startChatErrorHandler.ts

```diff
@@ -0,0 +1,134 @@
+import { ChatSDKErrorName, ChatSDKError } from "@microsoft/omnichannel-chat-sdk";
+import { LogLevel, TelemetryEvent } from "../../../common/telemetry/TelemetryConstants";
+import { TelemetryHelper } from "../../../common/telemetry/TelemetryHelper";
+import { TelemetryTimers } from "../../../common/telemetry/TelemetryManager";
+import { ConversationState } from "../../../contexts/common/ConversationState";
+import { LiveChatWidgetActionType } from "../../../contexts/common/LiveChatWidgetActionType";
+import { Dispatch } from "react";
+import { ILiveChatWidgetAction } from "../../../contexts/common/ILiveChatWidgetAction";
+import { callingStateCleanUp, endChatStateCleanUp, closeChatStateCleanUp, chatSDKStateCleanUp } from "./endChat";
+import { NotificationHandler } from "../../webchatcontainerstateful/webchatcontroller/notification/NotificationHandler";
+import { NotificationScenarios } from "../../webchatcontainerstateful/webchatcontroller/enums/NotificationScenarios";
+import { DataStoreManager } from "../../../common/contextDataStore/DataStoreManager";
+import { ILiveChatWidgetProps } from "../interfaces/ILiveChatWidgetProps";
+import { getWidgetCacheIdfromProps } from "../../../common/utils";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const handleStartChatError = (dispatch: Dispatch<ILiveChatWidgetAction>, chatSDK: any, props: ILiveChatWidgetProps | undefined, ex: any, isStartChatSuccessful: boolean) => {
+    if (!ex) {
+        logWidgetLoadFailed();
+        return;
+    }
+
+    if (ex instanceof ChatSDKError) {
+        if (ex.message === ChatSDKErrorName.WidgetUseOutsideOperatingHour) {
+            dispatch({ type: LiveChatWidgetActionType.SET_OUTSIDE_OPERATING_HOURS, payload: true });
+            dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.OutOfOffice });
+            logWidgetLoadComplete("Widget is OOOH.");
+            return;
+        } else if (ex.message === ChatSDKErrorName.PersistentChatConversationRetrievalFailure ||
```

Comment:
Would having an array of `ChatSDKErrorName` then use `ChatSDKErrorNamesWithHTTPResponseStatusCode.includes(ex.message)` to validate the message be more simple?

Reply:
Prefer to leave it like this, since the handling for any of these events might differ in the future (currently it happens to be "400 against everything else" for these 4 events, but we might add 429 in the future for sessionInit for example). Using an array adds unnecessary complexity.